### PR TITLE
added host volumes

### LINKED ISSUE #10: Backup/recovery mechanisms.

### LINKED ISSUE #12: Add host volume
- Add host volume
- Add a switch for host volume!

### PR DESCRIPTION
**Pull request by dangernil**
Closes #12 and #24 

- Added host volumes to store postgres data
- Added switch to host volumes
- Added dynamic variables
- Updated README

### CODE REVIEW

**Comment by zhenik** [comment]
Tests?

**Comment by claesgill** [comment]
Changed to draft since we have some issues with `permission denied` when setting `/var/lib/postgres/data` as a volume to `/vagrant/persitence/postgres`.

**Comment by claesgill** [comment]
## STATUS:
We're taking a break from this issue due to problems with permission denied with postgres and vagrant. This will unfortunately not be able to reach the release v0.1.0 deadline (02.10.2020).

Going to pick this up on monday next week.

**Comment by Neha-Sinha2305** [comment]
You have missed out this piece of code that we need to add so that the new host_policy in nomad gets applied. Without this you will end up getting 403 permission denied when run with **nomad_acl=true** :
```
- name: Fetch bootstrap token nomad
  shell:
    cmd: vault kv get -field=secret-id secret/nomad-bootstrap-token
  when: lookup('env', 'nomad_acl') | bool
  register: bootstrap_token

- name: Copy nomad policy file to /etc/nomad.d/policies/
  copy:
    src: host_volume_policy.hcl
    dest: /etc/nomad.d/policies/host_volume_policy.hcl
  when: lookup('env', 'nomad_acl') | bool

- name: Update default write policy on nomad
  shell:
    cmd: nomad acl policy apply write-default /etc/nomad.d/policies/host_volume_policy.hcl
  when: lookup('env', 'nomad_acl') | bool
  environment:
    NOMAD_TOKEN: "{{ bootstrap_token.stdout }}
```
You can check the code in [minio_repo](https://github.com/fredrikhgrelland/terraform-nomad-minio/blob/master/dev/vagrant/conf/post_bootstrap/01-update-nomad-write-default-policy.yml)

**Comment by claesgill** [comment]
> You have missed out this piece of code that we need to add so that the new host_policy in nomad gets applied. Without this you will end up getting 403 permission denied when run with **nomad_acl=true** :
> 
> ```
> - name: Fetch bootstrap token nomad
>   shell:
>     cmd: vault kv get -field=secret-id secret/nomad-bootstrap-token
>   when: lookup('env', 'nomad_acl') | bool
>   register: bootstrap_token
> 
> - name: Copy nomad policy file to /etc/nomad.d/policies/
>   copy:
>     src: host_volume_policy.hcl
>     dest: /etc/nomad.d/policies/host_volume_policy.hcl
>   when: lookup('env', 'nomad_acl') | bool
> 
> - name: Update default write policy on nomad
>   shell:
>     cmd: nomad acl policy apply write-default /etc/nomad.d/policies/host_volume_policy.hcl
>   when: lookup('env', 'nomad_acl') | bool
>   environment:
>     NOMAD_TOKEN: "{{ bootstrap_token.stdout }}
> ```
> 
> You can check the code in [minio_repo](https://github.com/fredrikhgrelland/terraform-nomad-minio/blob/master/dev/vagrant/conf/post_bootstrap/01-update-nomad-write-default-policy.yml)

Oh, that's right! Thank's! :pray:

File: variables.tf [hunk 1]
```diff
@@ -51,3 +56,14 @@ variable "container_environment_variables" {
   default     = ["PGDATA=/var/lib/postgresql/data"]
 }
 
+variable "volume_destination" {
+  type        = string
+  description = "Postes volume destination"
```
Comment (Neha-Sinha2305) [inline variables.tf]:
```suggestion
  description = "Postgres volume destination"
```
;-)

**Review by zhenik — APPROVED** [review]

**Review by Neha-Sinha2305 — APPROVED** [review]

**Pull request merged**